open_humans/account_views.py

```diff
@@ -97,14 +89,14 @@ def generate_username(self, form):
         )
 
 
-class MemberChangeEmailView(PrivateMixin, AccountSettingsView):
+class MemberChangeEmailView(PrivateMixin, AllauthEmailView):
     """
     Creates a view for the current user to change their email.
 
-    This is an email-only subclass of account's SettingsView.
+    Uses our own email change template.
     """
 
-    form_class = MemberChangeEmailForm
+    form_class = AllauthAddEmailForm
     template_name = 'member/my-member-change-email.html'
     success_url = reverse_lazy('my-member-settings')
```

Comment:
Despite this here, it seems that submitting this form returns the user to the same form, instead of going to `my-member-settings`.

Reply:
fixed